### DISCUSSION TITLE
MatchContext.RegexpCaptureGroups empty with glob Match strategy behavior ?

### DISCUSSION THREAD

**Discussion by jeanmichelcraye** [ory/oathkeeper#933] [Q&A] [answered]
May be I missed something in my rule config but I need to retreive element of the request to forward it to KETO authorizer but...

using here under rule with glob :
 **% curl  http://localhost:4455/foo/bar**
return :
MatchContext.RegexpCaptureGroups as empty as show X-Grousp header that display the feedback
;(

```yml
- id: allow-test
  upstream:
    url: http://host.docker.internal:8080/
    preserve_host: true
  match:
    url: http://localhost:4455/<{*,*/*}>
    methods:
    - GET
  authenticators:
  - handler: noop
  authorizer:
    handler: allow
  mutators:
  - handler: header
    config:
      headers:
        X-Groups: "{{ printIndex .MatchContext.RegexpCaptureGroups 0  }}"
`
        
 ----- Request Start ----->
Method:          GET
Path:            /foo/bar

Headers:
X-Groups:        
X-Header:        map[Accept:[*/*] User-Agent:[curl/7.77.0]]
X-Method:        GET
X-Url:           http://localhost:4455/foo/bar  
<----- Request End -----

**Comment by gen1us2k** [comment, marked as answer]
Yes. You're right. Oathkeeper does not support using .MatchContext.RegexpCaptureGroups using glob matching strategy.
https://github.com/ory/oathkeeper/blob/69ad28f87fd4f49e1ca5ed0d4bd1099b61a795a5/rule/engine_glob.go#L35 
```
// FindStringSubmatch is noop for now and always returns an empty array
```
Consider using regexp matching strategy

**Reply by vinckr** [reply]
Hey @jeanmichelcraye, just checking in if your problem was solved. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
